main/src/ca/uwaterloo/flix/language/phase/Stratifier.scala

```diff
@@ -43,7 +45,7 @@ object Stratifier extends Phase[Root, Root] {
   /**
     * A type alias for the stratification cache.
     */
-  type Cache = mutable.Map[Set[Name.Pred], Ast.Stratification]
+  type Cache = mutable.Map[Map[Name.Pred, Int], Ast.Stratification]
```

Comment:
I am just wondering whether this should be specific to the arity, if we were working towards something more general.

Reply:
I think that the cache should follow the specific implementation. If the cache has the types then it will contain multiple entries that have identical computation.

If you want to change the stratification you only has to change the cache and the equality function.